Scripts/RemediationScripts/Remediate-CdnFrontDoorMinTLSVersion.ps1

```diff
@@ -0,0 +1,1094 @@
+                                                                ![String]::IsNullOrWhiteSpace($_.isMinTLSVersionSetOnCustomDomain)
+                                                                ![String]::IsNullOrWhiteSpace($_.CertificateType)
+                                                            }
+
+    $totalFrontDoors = $(($validFrontDoorDetails|Measure-Object).Count)
+
+    if ($totalFrontDoors -eq 0)
+    {
+        Write-Host "No Front door found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        return
+    }
+
+    Write-Host "Found [$($totalFrontDoors)] Front Door(s)." -ForegroundColor $([Constants]::MessageType.Update)
+    Write-Host $([Constants]::SingleDashLine)
+    # Back up snapshots to `%LocalApplicationData%'.
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\SetMinTLSVersionForCdnFrontDoors"
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+
+    Write-Host "Minimum TLS Version will be reset on the following Front Door(s):"
+    $validFrontDoorDetails | Select-Object @{N="Resource Group Name"; E={$_.ResourceGroupName}},@{N="Resource Name"; E={$_.ResourceName}}, @{N="Domain Name"; E={$_.DomainName}},@{N="Certificate Type"; E={$_.CertificateType}} | Format-Table -AutoSize -Wrap
+    Write-Host $([Constants]::SingleDashLine)
+
+    if (-not $Force)
+    {
+        Write-Host "Do you want to reset minimum TLS Version for all Front Door? " -NoNewline
+            
+        $userInput = Read-Host -Prompt "(Y|N)"
+        Write-Host $([Constants]::SingleDashLine)
+        if($userInput -ne "Y")
+        {
+            Write-Host "Minimum TLS Version will not be reset for any Front Door. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+            Write-Host $([Constants]::SingleDashLine)
+            return
+        }
+        Write-Host "User has provided consent to reset minimum TLS Version on all Front Doors." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::SingleDashLine)
+    }
+    else
+    {
+        Write-Host "'Force' flag is provided. Minimum TLS Version will be reset on all Front Doors without any further prompts." -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+        Write-Host $([Constants]::SingleDashLine)
+    }
+
+    Write-Host "[Step 3 of 3] Reset Minimum TLS Version for Front Doors" 
+    Write-Host $([Constants]::SingleDashLine)
+
+    # Includes Front Doors, to which, previously made changes were successfully rolled back.
+    $frontDoorsRolledBack = @()
+
+    # Includes Front Doors that were skipped during roll back. There were errors rolling back the changes made previously.
+    $frontDoorsSkipped = @()
+
+    $requiredMinTLSVersion = 'TLS10'
```

Comment:
Hardcoded value might not work if they have tls1.1 previously, better we should save current TLS state while remediating and use same here again.

Reply:
Fixed. Now storing and configuring previous MinTLSVersion.